tracer/src/Datadog.Trace.SourceGenerators/Helpers/DiagnosticInfo.cs

```diff
@@ -10,12 +10,17 @@ namespace Datadog.Trace.SourceGenerators.Helpers;
 internal sealed record DiagnosticInfo
 {
     public DiagnosticInfo(DiagnosticDescriptor descriptor, Location? location)
```

Comment:
If I understand the change correctly, the [Microsoft.CodeAnalysis.Location type](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.location?view=roslyn-dotnet-4.7.0) is not equatable, so instead we're creating a record with the information we need. As a result, since the [Descriptor property](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.diagnosticdescriptor?view=roslyn-dotnet-4.7.0) and the Location property are now equatable, the entire DiagnosticInfo record can be cached.